src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpOverallOutcomeController.cs

```diff
@@ -0,0 +1,239 @@
+
+
+
+
+        [HttpGet]
+        [Authorize(Policy = "AccessApplication")]
+        public async Task<IActionResult> ApplicationApproved(Guid applicationId)
+        {
+            var model = await BuildApplicationSummaryViewModel(applicationId);
+            return View("~/Views/Roatp/ApplicationApproved.cshtml", model);
+        }
+
+        [HttpGet]
+        [Authorize(Policy = "AccessApplication")]
+        public async Task<IActionResult> FeedbackAdded(Guid applicationId)
+        {
+            var model = await BuildApplicationSummaryViewModel(applicationId);
+            return View("~/Views/Roatp/FeedbackAdded.cshtml", model);
+        }
+
+        [HttpGet]
+        [Authorize(Policy = "AccessApplication")]
+        public async Task<IActionResult> ApplicationApprovedAlreadyActive(Guid applicationId)
+        {
+            var model = await BuildApplicationSummaryViewModel(applicationId);
+            return View("~/Views/Roatp/ApplicationApprovedAlreadyActive.cshtml", model);
+        }
+
+        public async Task<IActionResult> DownloadFile(Guid applicationId, Guid sectionId,
+            string pageId, string questionId, string filename)
+        {
+            var response = await _qnaApiClient.DownloadFile(applicationId, sectionId, pageId, questionId,
+                filename);
+
+            if (response.IsSuccessStatusCode)
+            {
+                var fileStream = await response.Content.ReadAsStreamAsync();
+
+                return File(fileStream, response.Content.Headers.ContentType.MediaType, filename);
+            }
+
+            return NotFound();
+        }
+
+
+        private async Task<ApplicationSummaryViewModel> BuildApplicationSummaryViewModel(Guid applicationId)
+        {
+            var application = await _apiClient.GetApplication(applicationId);
+            var applicationData = application.ApplyData.ApplyDetails;
+
+            var model = new ApplicationSummaryViewModel
+            {
+                ApplicationId = application.ApplicationId,
+                UKPRN = applicationData.UKPRN,
+                OrganisationName = applicationData.OrganisationName,
+                TradingName = applicationData.TradingName,
+                ApplicationRouteId = applicationData.ProviderRoute.ToString(),
+                ApplicationReference = applicationData.ReferenceNumber,
+                SubmittedDate = applicationData?.ApplicationSubmittedOn,
+                ExternalComments = application?.ApplyData?.GatewayReviewDetails?.ExternalComments,
+                EmailAddress = User.GetEmail(),
+                FinancialReviewStatus = application?.FinancialReviewStatus,
+                FinancialGrade = application?.FinancialGrade?.SelectedGrade,
+                FinancialExternalComments = application?.FinancialGrade?.ExternalComments,
+                GatewayReviewStatus = application?.GatewayReviewStatus,
+                ModerationStatus = application?.ModerationStatus
+            };
```

Comment:
TODO: Requires the fixes implemented in APR-2509

Reply:
ok, I lifted the code in, specifically line: ExternalComments = application.ExternalComments ?? application.ApplyData.GatewayReviewDetails?.ExternalComments,